### DISCUSSION TITLE
Coding style and performance

### DISCUSSION THREAD

**Discussion by oaklandgit** [replit/kaboom#415] [Q&A] [answered]
Hi,

I'm wondering if there are any downsides (e.g. performance impact) to this first approach (with the `add()` function embedded)…

```js
const player = add([
	sprite("bean"),
	pos(120, 80),
	rotate(0),
	origin("center"),
	{
		add() {
			this.onUpdate(() => {
				this.angle += 120 * dt()
			})
		}
	}
	
])
```

as opposed to the approach provided in most of the demos and documentation:

```js
const player = add([
	sprite("bean"),
	pos(120, 80),
	rotate(0),
	origin("center"),
])

player.onUpdate(() => {
	player.angle += 120 * dt()
})
```

The reason I prefer the former is that it encapsulates the behavior with the object. In testing a trivial example like the one above, there doesn't seem to be a performance impact but I wonder if there could be theoretically or any other reason it should be avoided.

Thank you

**Comment by slmjkdbtl** [comment, marked as answer]
They're exactly the same, although for the first one I'd write it in
```js
add([
    {
        update() {
            // ...
        },
    },
])
```
instead of `onUpdate()` in `add()` to be more succinct.

**Reply by oaklandgit** [reply]
Ah even better. Thank you!
